packages/gestalt/src/contexts/Theme.jsdom.test.js

```diff
@@ -0,0 +1,159 @@
+// @flow strict
+import React from 'react';
+import { act, render } from '@testing-library/react';
+import { ThemeProvider, useTheme } from './Theme.js';
+
+function ThemeAwareComponent() {
+  const theme = useTheme();
+  return <div>{theme.name}</div>;
+}
+
+describe('Themeing', () => {
+  describe('ThemeProvider', () => {
+    it('renders child content in a div with a unique className', () => {
+      const { container } = render(
+        <>
+          <ThemeProvider>Child 1</ThemeProvider>
+          <ThemeProvider>Child 2</ThemeProvider>
+        </>
+      );
+      expect(container.querySelectorAll('div')).toMatchInlineSnapshot(`
+        NodeList [
+          <div
+            class="__gestaltTheme1"
+          >
+            Child 1
+          </div>,
+          <div
+            class="__gestaltTheme2"
+          >
+            Child 2
+          </div>,
+        ]
+      `);
+    });
+    it('renders styling for light mode when no color scheme specified', () => {
+      const { container } = render(<ThemeProvider>Content</ThemeProvider>);
+      expect(container.querySelectorAll('style')).toMatchInlineSnapshot(`
+        NodeList [
+          <style>
+            .__gestaltTheme3 {
+          --colorRed0: #ff5247;
```

Comment:
Should these be `--color-red-0`? Seems like it might be more "CSS Native" (though don't feel strongly about it).

Reply:
Hmm, could go either way. I like that this mostly matches the names we use in JS like `const { colorGray100 } = useTheme();` would look weird in the more css convention though they don't have to match

Reply:
@jennyscript please prefix all of these with `--gestalt-`. See #971 for more context

Reply:
Updated! That did increase the filesize a bit but since it's repetitive it should be very gzip-able